common-libs/common/src/services/DidDocumentService/index.ts

```diff
@@ -9,40 +10,24 @@ export { KeyVault } from './KeyVault'
 export { LocalKeyVault } from './LocalKeyVault'
 
 export default class DidDocumentService {
+  /**
+   * @deprecated use DidDocumentService.createDidDocumentService instead
+   */
   constructor(keysService: KeysService) {
-    const { didMethod } = keysService.decryptSeed()
-
-    let didDocumentService
-    switch (didMethod) {
-      case 'jolo':
-        didDocumentService = new JoloDidDocument(keysService)
-        break
-      case 'elem':
-        didDocumentService = new ElemDidDocument(new LocalKeyVault(keysService))
-        break
-    }
-
-    return didDocumentService
-  }
-
-  getMyDid() {
-    return 'did:...'
+    return DidDocumentService.createDidDocumentService(keysService)
   }
 
-  getKeyId(did: string = null) {
-    if (!did) {
-      did = this.getMyDid()
-    }
-
-    const signingKey = 'primary'
-
-    return `${did}#${signingKey}`
-  }
+  static createDidDocumentService(
+    keysService: KeysService,
+  ): JoloDidDocument | ElemDidDocument | ElemAnchoredDidDocument {
```

Comment:
By the way TS should be able to automatically infer the type there.
Or you can use `ConcreteDidDocument`.